<launch>
  <arg name="input" default="points" />

  <!-- launch the kinect -->
  <include file="$(find openni_launch)/launch/openni.launch"/>

  <!-- launch the nodelet manager -->
  <node pkg="nodelet" type="nodelet" name="pcl_manager" args="manager" output="screen"/>  

  <!-- tracker nodes and static frame broadcaster -->
  <node pkg="pend_tracker" type="pend_track"
  	name="pend_tracking_node" output="screen" /> -->

  <node pkg="tf" type="static_transform_publisher" name="optimization_frame_broadcaster" 
args="0 0 0 -0.0018491 -0.187725 0.98222 -0.000353407 camera_depth_optical_frame oriented_optimization_frame 100" />

  <!-- voxel downsampling, and transformation -->
  <group ns="/cloud_downsampler">
    <node pkg="nodelet" type="nodelet" name="voxel" args="load pcl/VoxelGrid /pcl_manager" output="screen">  
      <remap from="~input" to="/camera/depth/$(arg input)" />
      <rosparam>
        filter_field_name: z
        filter_limit_min: 0
        filter_limit_max: 2.5
        fiter_limit_negative: True
        leaf_size: 0.01
        output_frame: /oriented_optimization_frame
      </rosparam>
    </node>
  </group>
  
  <!-- pass-through filtering for the mass:-->  
  <group ns="/mass_box_filter">
    <!-- Run a passthrough filter to delimit in x direction -->
    <node pkg="nodelet" type="nodelet" name="psx" args="load pcl/PassThrough /pcl_manager" output="screen">
      <remap from="~input" to="/cloud_downsampler/voxel/output" />
      <rosparam>
        filter_field_name: x
        filter_limit_min: -1.0
        filter_limit_max: 1.0
      </rosparam>
    </node>
    <!-- Run a passthrough filter to delimit in y direction -->
    <node pkg="nodelet" type="nodelet" name="psy" args="load pcl/PassThrough /pcl_manager" output="screen">
      <remap from="~input" to="psx/output" />
      <rosparam>
        filter_field_name: y
        filter_limit_min: -1.45
        filter_limit_max: 0.0
      </rosparam>
    </node>
  </group>

</launch>
